src/tree/updater_gpu_hist.cu

```diff
@@ -626,6 +631,14 @@ struct GPUHistMakerDevice {
     return std::vector<DeviceSplitCandidate>(result_all.begin(), result_all.end());
   }
 
+  // Build gradient histograms for a given node across all the batches in the DMatrix.
+  void BuildHistBatches(int nidx, DMatrix* p_fmat) {
+    for (auto& batch : p_fmat->GetBatches<EllpackPage>(batch_param)) {
```

Comment:
wouldn't this result in pages getting recycled aggressively while walking through the batches, thus resulting in ellpack pages repeatedly getting created and destructed leading to large number of data copies to device and removal from it?